### PR TITLE
Chore: Convert CarouselInput to Typescript

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

Convert `CarouselInput` to Typescript.

### Why is it needed?

https://github.com/strapi/design-system/issues/983

### CODE REVIEW

File: packages/strapi-design-system/src/CarouselInput/CarouselImage.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    setIsError(true);
+  };
+
+  if (isError) {
+    return (
+      <Tooltip description={props.alt ?? ''}>
+        <StyledImage as="img" height="100%" {...props} />
+      </Tooltip>
+    );
+  }
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/CarouselInput/CarouselImage.tsx:31]: I have no understand why we do this – do you? 😓

File: packages/strapi-design-system/src/CarouselInput/CarouselSlide.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Flex, FlexProps } from '../Flex';
+
+export interface CarouselSlideProps extends FlexProps {
+  children: React.ReactNode;
+  label: string;
+}
+
+export const CarouselSlide = ({ label, children, selected = false, ...props }) => (
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/CarouselInput/CarouselSlide.tsx:8]:
```suggestion
export const CarouselSlide = ({ label, children, selected = false, ...props }: CarouselSlideProps) => (
```

File: packages/strapi-design-system/src/CarouselInput/Carousel.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+export interface CarouselProps extends BoxProps {
+  actions?: React.ReactNode;
+  children: React.ReactNode;
+  label: string;
+  nextLabel: string;
+  onNext: () => void;
+  onPrevious: () => void;
+  previousLabel: string;
+  secondaryLabel: React.ReactNode;
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/CarouselInput/Carousel.tsx:21]: I think this is optional looking at the code

File: packages/strapi-design-system/src/CarouselInput/Carousel.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  children,
+  label,
+  nextLabel,
+  onNext,
+  onPrevious,
+  previousLabel,
+  secondaryLabel,
+  selectedSlide,
+  ...props
+}) => {
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/CarouselInput/Carousel.tsx:54]:
```suggestion
}: CarouselProps) => {
```

File: packages/strapi-design-system/src/CarouselInput/CarouselImage.tsx (hunk 1 above)
Comment (joshuaellis) [reply packages/strapi-design-system/src/CarouselInput/CarouselImage.tsx:31]: Nah it's fine, if you don't know it off the top of your head I can try and live with it 🥲

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

File: packages/strapi-design-system/src/CarouselInput/CarouselActions.tsx [hunk 5]
```diff
@@ -1,8 +1,16 @@
...
-import { Flex } from '../Flex';
+export interface CarouselActionsProps extends FlexProps {
+  horizontal?: boolean;
 
-export const CarouselActions = ({ horizontal = true, spacing = 1, ...props }) => (
+  /**
+   * @preserve
+   * @deprecated use `gap` instead
+   */
+  spacing?: FlexProps['gap'];
```
[the remark this answers is not in the document]
Comment (HichamELBSI) [reply packages/strapi-design-system/src/CarouselInput/CarouselActions.tsx:10]: Oops my bad, it is already handled as the props are spread to the Flex component 🙏🏽

**Review by HichamELBSI — APPROVED** [review]
Good job!

**Review by joshuaellis — APPROVED** [review]
💯 nice

**Pull request merged**